hr_contract_cae/models/hr_contract.py

```diff
@@ -19,20 +144,162 @@ class Contract(models.Model):
         help="Employee's monthly gross wage.",
         compute="_compute_wage",
     )
+    notes = fields.Text(copy=False)
+
+    attachment_number = fields.Integer(
+        compute="_compute_attachment_number", string="Number of Attachments"
+    )
+    attachment_ids = fields.One2many(
+        comodel_name="ir.attachment",
+        inverse_name="res_id",
+        string="Attachments",
+        domain=[("res_model", "=", "hc.contract")],
+    )
+
+    @api.model
+    def create(self, vals):
+        res = super().create(vals)
+        if not res.contract_group_id:
+            contract_group = self.env["hr.contract.group"].create({})
+            res.contract_group_id = contract_group
+        if res.amendment_index == 0:
+            res.initial_contract_id = res.get_initial()
```

Comment:
Would not bee needed if the `initial_contract_id` field was defined on the group. Plus, if `res.amendment_index == 0`, you already know that `res.initial_contract_id = self`.